File: Projekt/src/main/java/hr/algebra/java/web/projekt/projekt/models/Member.java
package hr.algebra.java.web.projekt.projekt.models;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Data
@Table(name = "Members")
public class Member {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int idMembers;

    @Column(nullable = false, length = 255)
    private String firstName;

    @Column(nullable = false, length = 255)
    private String lastName;

    @Column(length = 255)
    private String address;

    @Column(length = 11)
    private String croatianPIN;

    public String getFullName() {
        return firstName + " " + lastName;
    }
    @ManyToOne
    @JoinColumn(name = "memberTypeId", nullable = false)
    private MemberType memberType;

    @Column(nullable = false, length = 255)
    private String email;

    @Column(nullable = false, length = 1)
    private String gender; // 'M', 'F' ili 'N'

    @Enumerated(EnumType.ORDINAL)
    @Column(nullable = false)
    private PackageType packageType;
}

File: angular-fontend/src/app/models/member.ts
import { PackageType } from "./package-type";
import { MemberType } from "./member-type";


export class Member {
  idMembers: number;
  fullName: string;
  address:string;
  croatianPIN: string;
  memberType: MemberType;
  email: string;
  gender: 'M' | 'F' | 'N';
  packageType: PackageType;

  constructor(
    idMembers: number,
    fullName: string,
    address: string,
    croatianPIN: string,
    memberType: MemberType,
    email: string,
    gender: 'M' | 'F' | 'N',
    packageType: PackageType
  ) {
    this.idMembers = idMembers;
    this.fullName = fullName;
    this.address=address;
    this.croatianPIN = croatianPIN;
    this.memberType = memberType;
    this.email = email;
    this.gender = gender;
    this.packageType = packageType;
  }
}
